crates/optimism/node/src/txpool.rs

```diff
@@ -309,6 +321,37 @@ where
             )
         }
 
+        // If the EVM is configured, enable interop tx execution.
+        if let Some(evm_cfg) = &self.evm {
+            // Get the current block
+            let block = if let Ok(Some(block)) = self.inner.client().block_by_number_or_tag(alloy_eips::BlockNumberOrTag::Latest) {
+                block
+            } else {
+                // Failure to get the latest block is a critical error.
+                // Transaction validation must be re-tried.
+                // TODO: fix this error we need a custom type
+                return TransactionValidationOutcome::Invalid(
+                    transaction,
+                    InvalidTransactionError::TxTypeNotSupported.into(),
+                )
+            };
+
+            // Construct the state using the current block.
+            let state = self.inner.client().state_by_block_hash(block.hash())?;
+            let mut db =
+                CacheDB::new(StateProviderDatabase::new(&state));
+
+            // Configure the evm
+            let mut evm = evm_cfg.evm_for_block(&mut db, &block.header());
+
+            // Construct the environment for the transaction
+            // let sender = transaction.sender();
+            // let mut tx_env = evm_cfg.tx_env(transaction, *sender);
+
+            // Transact
+            let _ = evm.transact();
```

Comment:
```suggestion
            // Construct the environment for the transaction
            // let sender = transaction.sender();
            // let mut tx_env = evm_cfg.tx_env(transaction, *sender);

            // Transact
            let ResultAndState { result: _, .. } = evm.transact(tx_env);
```
you find the logs on the result, but we're not using them in this step. this is issue is just ab executing transactions (and discarding those that fail execution)